<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>896</width>
    <height>303</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>hasaki</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QTableWidget" name="mappingsTableWidget"/>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QLabel" name="processPresetLabel">
        <property name="text">
         <string>进程预设:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="processPresetComboBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>150</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>150</width>
          <height>16777215</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="addProcessPresetButton">
        <property name="maximumSize">
         <size>
          <width>30</width>
          <height>30</height>
         </size>
        </property>
        <property name="toolTip">
         <string>新增进程预设</string>
        </property>
        <property name="text">
         <string>+</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="editProcessPresetButton">
        <property name="maximumSize">
         <size>
          <width>30</width>
          <height>30</height>
         </size>
        </property>
        <property name="toolTip">
         <string>编辑当前预设</string>
        </property>
        <property name="text">
         <string>✏️</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="copyProcessPresetButton">
        <property name="maximumSize">
         <size>
          <width>30</width>
          <height>30</height>
         </size>
        </property>
        <property name="toolTip">
         <string>复制当前预设</string>
        </property>
        <property name="text">
         <string>📋</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="deleteProcessPresetButton">
        <property name="maximumSize">
         <size>
          <width>30</width>
          <height>30</height>
         </size>
        </property>
        <property name="toolTip">
         <string>删除当前预设</string>
        </property>
        <property name="text">
         <string>❌</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="upstreamLabel">
        <property name="text">
         <string>上游服务器:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QComboBox" name="upstreamComboBox">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>200</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>200</width>
          <height>16777215</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="editUpstreamButton">
        <property name="maximumSize">
         <size>
          <width>30</width>
          <height>30</height>
         </size>
        </property>
        <property name="toolTip">
         <string>编辑当前服务器</string>
        </property>
        <property name="text">
         <string>✏️</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="deleteUpstreamButton">
        <property name="maximumSize">
         <size>
          <width>30</width>
          <height>30</height>
         </size>
        </property>
        <property name="toolTip">
         <string>删除当前服务器</string>
        </property>
        <property name="text">
         <string>❌</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Orientation::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="startButton">
        <property name="text">
         <string>启动</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="stopButton">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>停止</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>896</width>
     <height>20</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>选项</string>
    </property>
    <addaction name="actionConsoleWindow"/>
    <addaction name="actionSettings"/>
    <addaction name="actionAddUpstream"/>
    <addaction name="actionUdpTest"/>
   </widget>
   <widget class="QMenu" name="menu_2">
    <property name="title">
     <string>帮助</string>
    </property>
   </widget>
   <addaction name="menu"/>
   <addaction name="menu_2"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionConsoleWindow">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>打开控制台</string>
   </property>
  </action>
  <action name="actionSettings">
   <property name="text">
    <string>设置</string>
   </property>
  </action>
  <action name="actionAddUpstream">
   <property name="text">
    <string>添加上游</string>
   </property>
   <property name="toolTip">
    <string>添加上游</string>
   </property>
  </action>
  <action name="actionUdpTest">
   <property name="text">
    <string>UDP测试工具</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
